### ISSUE TITLE
After upgrading from gloo-ee version 1.10.10 to 1.11.6 proxy resource is no longer updated  

### ISSUE THREAD

**Issue by avizov** [solo-io/gloo#6406]
### Gloo Edge Version

1.11.x (latest stable)

### Kubernetes Version

1.21.x

### Describe the bug

We upgraded the gloo-ee chart version from 1.10.10 to 1.11.6 and the proxy resource is no longer updated with a new configuration. In the gateway pod we can see the following errors: 

{"level":"error","ts":"2022-05-02T09:28:12.458Z","logger":"gateway.v1.event_loop.gateway","caller":"syncer/setup_syncer.go:335","msg":"error: event_loop: 1 error occurred:\n\t* reconciling resource gateway-proxy: updating kube resource gateway-proxy: (want 10694): proxies.gloo.solo.io \"gateway-proxy\" is invalid: metadata.resourceVersion: Invalid value: 0x0: must be specified for an update\n\n","version":"1.11.6","stacktrace":"github.com/solo-io/gloo/projects/gateway/pkg/syncer.RunGateway.func1\n\t/workspace/gloo/projects/gateway/pkg/syncer/setup_syncer.go:335"}

### Steps to reproduce the bug

1. upgrade the gloo-ee
2. create a new virtual service 

### Expected Behavior

The proxy resource is updated with a new configuration

**Comment by avizov** [comment]
the same issue was observed in 1.10.x latest stable version as well
{"level":"error","ts":"2022-05-04T11:33:46.727Z","logger":"gateway.v1.event_loop.gateway","caller":"syncer/setup_syncer.go:335","msg":"error: event_loop: 1 error occurred:\n\t* reconciling resource gateway-proxy: updating kube resource gateway-proxy: (want 11424): proxies.gloo.solo.io \"gateway-proxy\" is invalid: metadata.resourceVersion: Invalid value: 0x0: must be specified for an update\n\n","version":"1.10.23","stacktrace":"github.com/solo-io/gloo/projects/gateway/pkg/syncer.RunGateway.func1\n\t/workspace/gloo/projects/gateway/pkg/syncer/setup_syncer.go:335"}

**Comment by chrisgaun** [comment]
Thanks @avizov. 

WE have a dupe issue of this somewhere.

**Comment by avizov** [comment]
any news?

**Comment by sam-heilbron** [comment]
@avizov , we have identified the source of the bug, have a temporary workaround and are actively working on a more robust solution that should be delivered in a couple of days.

Problem: Gloo maintains an internal CR, known as a [Proxy](https://github.com/solo-io/gloo/blob/e5df9c889c00a4756fc468432160732706554257/projects/gloo/api/v1/proxy.proto#L30). We uniquely identify these Proxies with [unique labels](https://github.com/solo-io/gloo/blob/bc380b36d42fdad7c83ab8dc4f055258b326aeac/projects/gateway/pkg/syncer/translator_syncer.go#L49). We rely on these labels to both read Proxies (to know which to reconcile) and write Proxies (to know which to reconcile on the next iteration). Recently we changed these labels on Proxies, which means that after an upgrade from a version with the old label, to a version with the new label, the Proxy controller will attempt to reconcile the existing persisted Proxies. However, since the Proxy uses the old label, but the new controller looks for the new label, it will find no proxies, and expect to create one. Our toolkit for modifying resources will then error, since it is attempting to create a resource that is meant for deletion.

Temporary Workaround #1: We previously recommended deleting the Proxy and allowing it to be regenerated. If you do this, you should consider using [disableProxyGarbageCollection](https://github.com/solo-io/gloo/blob/bc380b36d42fdad7c83ab8dc4f055258b326aeac/projects/gloo/api/v1/settings.proto#L549) to ensure that the xDS snapshot for that (now deleted) Proxy is maintained in the xds cache.

Temporary Workaround #2: Instead of deleting the Proxy, you could modify the label to match what our controllers expect. Doing so will avoid any potential for downtime.

Work In Progress Solution: There are a few options for how we can resolve this more thoroughly. Expanding our usage of [label selection](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/) to include set-based requirements is one option. We will decide internally on the direction and put up a more robust solution that doesn't require manual intervention soon. I will keep you updated on this issue.

**Comment by totallyGreg** [comment]
Duplicate of https://github.com/solo-io/gloo/issues/6322

**Comment by sam-heilbron** [comment]
This has been resolved in:
Gloo OSS - 1.11.11, 1.10.25, 1.9.15 
Gloo EE - 1.11.10, 1.10.23, 1.9.18

The fix is introduced into the 1.12 beta branch, and will be released in the coming days.

To verify, you can use the following steps:
```
# example: 1.10.0
GLOO_VERSION="${GLOO_VERSION}"

# create the cluster
kind create cluster --name "gloo-$GLOO_VERSION"

# install gloo
helm install gloo gloo/gloo --version "$GLOO_VERSION" -n gloo-system --create-namespace

# deploy petstore
kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo/v1.2.9/example/petstore/petstore.yaml

# add route
glooctl upgrade --release="v$GLOO_VERSION"

glooctl add route \
  --path-exact /all-pets \
  --dest-name default-petstore-8080 \
  --prefix-rewrite /api/pets

kubectl -n gloo-system rollout status deployment gateway-proxy --timeout=2m
kubectl -n gloo-system rollout status deployment gloo --timeout=2m
kubectl -n gloo-system rollout status deployment gateway --timeout=2m
kubectl -n gloo-system rollout status deployment discovery --timeout=2m

# example: 1.11.10
GLOO_UPGRADE_VERSION="${GLOO_UPGRADE_VERSION}"

# apply CRDs
helm pull gloo/gloo --version "$GLOO_UPGRADE_VERSION" --untar
kubectl apply -f gloo/crds
rm -rf gloo

# upgrade
helm upgrade gloo gloo/gloo --version "v$GLOO_UPGRADE_VERSION" --namespace gloo-system

```
